statuslistener/statuslistener.go

```diff
@@ -0,0 +1,153 @@
+package statuslistener
+
+import (
+	"sort"
+	"time"
+
+	c "github.com/RedHatInsights/sources-api-go/config"
+	"github.com/RedHatInsights/sources-api-go/dao"
+	"github.com/RedHatInsights/sources-api-go/internal/events"
+	"github.com/RedHatInsights/sources-api-go/kafka"
+	logging "github.com/RedHatInsights/sources-api-go/logger"
+	m "github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/util"
+)
+
+const (
+	SourcesStatusTopic      = "platform.sources.status"
+	GroupID                 = "sources-api-status-worker"
+	EventAvailabilityStatus = "availability_status"
+)
+
+var config = c.Get()
+
+type AvailabilityStatusListener struct {
+	*events.EventStreamProducer
+}
+
+func Run() {
+	avs := AvailabilityStatusListener{EventStreamProducer: NewEventStreamProducer()}
+	avs.subscribeToAvailabilityStatus()
+}
+
+func NewEventStreamProducer() *events.EventStreamProducer {
+	sender := &events.EventStreamSender{}
+	return &events.EventStreamProducer{Sender: sender}
+}
+
+func (avs *AvailabilityStatusListener) subscribeToAvailabilityStatus() {
+	if logging.Log == nil {
+		panic("logging is not initialized")
+	}
+
+	consumerConfig := kafka.ConsumerConfig{Topic: config.KafkaTopic(SourcesStatusTopic), GroupID: GroupID}
+	kafkaConfig := kafka.Config{KafkaBrokers: config.KafkaBrokers, ConsumerConfig: consumerConfig}
+
+	kf := &kafka.Manager{Config: kafkaConfig}
+
+	errorConsume := kf.Consume(avs.ConsumeStatusMessage)
+
+	if errorConsume != nil {
+		logging.Log.Errorf("Consumer kafka message error: %s", errorConsume.Error())
+	}
+}
+
+func (avs *AvailabilityStatusListener) ConsumeStatusMessage(message kafka.Message) {
+	var statusMessage StatusMessage
+	err := message.ParseTo(&statusMessage)
+	if err != nil {
+		logging.Log.Errorf("Error in parsing status message %s", err.Error())
```

Comment:
So for this kind of log its probably better to use `%v` (its automatic per the type) that way you only have to pass in `err` instead of calling `err.Error()`.